### ISSUE TITLE
Error when transcribing

### ISSUE THREAD

**Issue by pearl-yu** [pszemraj/vid2cleantxt#15]
Here it is:
UnboundLocalError: local variable 'PL_out' referenced before assignment

I was running the Colab notebook only changing the video URL to a google drive link.

**Comment by pszemraj** [comment]
hey, thanks for reporting that! Sorry, that error is somewhat vague. nothing was transcribed because it (likely) didn't find any videos to transcribe. Currently, no error is thrown when vid2cleantxt tries to transcribe a directory and finds nothing, but `UnboundLocalError: local variable 'PL_out' referenced before assignment` is thrown during post-processing (PL_out doesn't exist because nothing was transcribed)

## core problem/solution

- the vid2cleantxt notebook currently uses `wget ` to load videos. in `!wget -O $local_path $url_to_video` to download a video. [this does not work for google drive links](https://stackoverflow.com/a/50670037)
- instead, you probably want to use the `gdown` command (see the [gdown](https://github.com/wkentaro/gdown) repo. note that this requires you to know/figure out the drive url ID _and_ make that file has the permissions "anyone with the link can view"
- gdown comes already installed on colab so you don't need to do anything to load it

example:

- I want to download [this video on my drive](https://drive.google.com/file/d/1c-VJfq-qNSy8DQaxkx_yiopGidxhauGG/view?usp=sharing) with URL of `https://drive.google.com/file/d/1c-VJfq-qNSy8DQaxkx_yiopGidxhauGG/view?usp=sharing`
- the ID is after `/d/`: `1c-VJfq-qNSy8DQaxkx_yiopGidxhauGG`

in Colab:

```python
drive_id = "1c-VJfq-qNSy8DQaxkx_yiopGidxhauGG"
!gdown $drive_id 
```

so basically, replace `!wget -O $local_path $url_to_video` with the above (except for your file), and it should work. Outside of that, you can see [this medium post](https://towardsdatascience.com/different-ways-to-connect-google-drive-to-a-google-colab-notebook-pt-1-de03433d2f7a) or [this stack overflow](https://stackoverflow.com/questions/48376580/google-colab-how-to-read-data-from-my-google-drive) for other ways of loading video files to Colab

---

_EDIT: actually, you don't even need to provide the `-O` arg, so I removed it from the code block above._


![colab-screen](https://i.imgur.com/0xbxI0t.png)

**Comment by pszemraj** [comment]
hey! let me know if the above solved your problem or if you need further help. Otherwise, I'll close this issue in a few days :)
